Warning: token pasting of ',' and __VA_ARGS__ is a GNU extension

Hi, building with `juce_recommended_warning_flags` I get this warning:

```
_deps/melatonin_perfetto-src/melatonin_perfetto/melatonin_perfetto.h:249:69: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
            TRACE_EVENT ("dsp", perfetto::StaticString (pf.data()), ##__VA_ARGS__)
```

I had the same issue with my own logging macros and simply removing the `##` solved it. I'm building with Clang though, so I thought those might would be needed to make gcc happy, but [maybe not actually](https://godbolt.org/z/s96YMnTdv)...

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -32,6 +32,8 @@ jobs:
       fail-fast: false
       matrix:
         include:
+          - name: Linux
+            os: ubuntu-latest
           - name: macOS
             os: macos-latest
           - name: Windows

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -44,7 +44,7 @@ jobs:
       - name: Install JUCE's Linux Deps
         if: runner.os == 'Linux'
         run: |
-          sudo apt-get update && sudo apt install libasound2-dev libx11-dev libxinerama-dev libxext-dev libfreetype6-dev libwebkit2gtk-4.1-dev libglu1-mesa-dev xvfb ninja-build
+          sudo apt-get update && sudo apt install libasound2-dev libcurl4-openssl-dev libx11-dev libxinerama-dev libxext-dev libfreetype6-dev libwebkit2gtk-4.1-dev libglu1-mesa-dev xvfb ninja-build
           sudo /usr/bin/Xvfb $DISPLAY &
 
       - name: Checkout code

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -41,6 +41,12 @@ jobs:
 
     steps:
 
+      - name: Install JUCE's Linux Deps
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt-get update && sudo apt install libasound2-dev libx11-dev libxinerama-dev libxext-dev libfreetype6-dev libwebkit2gtk-4.1-dev libglu1-mesa-dev xvfb ninja-build
+          sudo /usr/bin/Xvfb $DISPLAY &
+
       - name: Checkout code
         uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c   # pin@v3
         with:

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,6 +59,7 @@ if (MSVC)
             /bigobj # only needed for compilation of perfetto itself
         PUBLIC
             /Zc:__cplusplus # we need the correct value of the __cplusplus macro
+            /Zc:preprocessor # needed for __VA_OPT__(,) support
             /permissive- # see https://github.com/google/perfetto/issues/214
         )
 endif ()

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.24 FATAL_ERROR)
 
 project(MelatoninPerfetto
-        VERSION 1.3.0
+        VERSION 1.4.0
         LANGUAGES CXX
         DESCRIPTION "JUCE module for profiling with Perfetto"
         HOMEPAGE_URL "https://github.com/sudara/melatonin_perfetto")
@@ -27,7 +27,7 @@ set(MP_INSTALL_DEST "${CMAKE_INSTALL_LIBDIR}/cmake/melatonin_perfetto"
 
 
 message (STATUS "Grabbing Perfetto...")
-CPMAddPackage(gh:google/perfetto@51.2)
+CPMAddPackage(gh:google/perfetto@53.0)
 
 # we need to manually set up a target for Perfetto
 add_library(perfetto STATIC)

--- a/README.md
+++ b/README.md
@@ -329,7 +329,7 @@ Go wild!
 
 ## Assumptions / Caveats
 
-* On Mac, the trace is dumped to your Downloads folder. On Windows, it's dumped to your Desktop (sorry not sorry).
+* On Mac, the trace is dumped to your Downloads folder. On Windows, it's dumped to your Desktop. On Linux, it's dumped to the temp directory (`/tmp`).
 * Traces are set to in memory, 80MB by default.
 
 ## Troubleshooting

--- a/melatonin_perfetto/melatonin_perfetto.h
+++ b/melatonin_perfetto/melatonin_perfetto.h
@@ -256,7 +256,7 @@ namespace melatonin
 #if PERFETTO_ENABLE_TRACE_DSP
     #define TRACE_DSP(...)                                                                                                            \
         static constexpr auto pf = melatonin::compileTimePrettierFunction (WRAP_COMPILE_TIME_STRING (PERFETTO_DEBUG_FUNCTION_IDENTIFIER())); \
-        TRACE_EVENT ("dsp", perfetto::StaticString (pf.data()), ##__VA_ARGS__)
+        TRACE_EVENT ("dsp", perfetto::StaticString (pf.data()) __VA_OPT__(,) __VA_ARGS__)
 
     #define TRACE_DSP_BEGIN(name) TRACE_EVENT_BEGIN ("dsp", perfetto::StaticString (name))
     #define TRACE_DSP_END() TRACE_EVENT_END ("dsp")
@@ -269,7 +269,7 @@ namespace melatonin
 #if PERFETTO_ENABLE_TRACE_COMPONENT
     #define TRACE_COMPONENT(...)                                                                                                      \
         static constexpr auto pf = melatonin::compileTimePrettierFunction (WRAP_COMPILE_TIME_STRING (PERFETTO_DEBUG_FUNCTION_IDENTIFIER())); \
-        TRACE_EVENT ("component", perfetto::StaticString (pf.data()), ##__VA_ARGS__)
+        TRACE_EVENT ("component", perfetto::StaticString (pf.data()) __VA_OPT__(,) __VA_ARGS__)
 
     #define TRACE_COMPONENT_BEGIN(name) TRACE_EVENT_BEGIN ("component", perfetto::StaticString (name))
     #define TRACE_COMPONENT_END() TRACE_EVENT_END ("component")

--- a/melatonin_perfetto/melatonin_perfetto.h
+++ b/melatonin_perfetto/melatonin_perfetto.h
@@ -95,8 +95,11 @@ public:
     {
     #if JUCE_WINDOWS
         return juce::File::getSpecialLocation (juce::File::SpecialLocationType::userDesktopDirectory);
-    #else
+    #elif JUCE_MAC || JUCE_IOS
         return juce::File::getSpecialLocation (juce::File::SpecialLocationType::userHomeDirectory).getChildFile ("Downloads");
+    #else
+        // Linux: use temp directory for CI compatibility (no desktop/Downloads on headless systems)
+        return juce::File::getSpecialLocation (juce::File::SpecialLocationType::tempDirectory);
     #endif
     }
 

--- a/scripts/RunTests.py
+++ b/scripts/RunTests.py
@@ -30,7 +30,8 @@ def run_command(command, workingDir):
         print(cp.stderr)
 
     except subprocess.CalledProcessError as error:
-        print (error.output)
+        print("STDOUT:", error.stdout)
+        print("STDERR:", error.stderr)
         print (f"Command {error.cmd} failed with exit code {error.returncode}")
         exit (1)
 
@@ -55,7 +56,7 @@ run_command (command=f"cmake -B {BUILD_DIR}{build_type_flag}",
 run_command (command=f"cmake --build {BUILD_DIR}",
              workingDir=REPO_ROOT)
 
-run_command (command="ctest -C Debug",
+run_command (command="ctest -C Debug --output-on-failure",
              workingDir=BUILD_DIR)
 
 environ["MP_PERFETTO_SHOULD_BE_ON"] = "TRUE"
@@ -66,5 +67,5 @@ run_command (command=f"cmake -B {BUILD_DIR} -D PERFETTO=ON{build_type_flag}",
 run_command (command=f"cmake --build {BUILD_DIR}",
              workingDir=REPO_ROOT)
 
-run_command (command="ctest -C Debug",
+run_command (command="ctest -C Debug --output-on-failure",
              workingDir=BUILD_DIR)

--- a/scripts/RunTests.py
+++ b/scripts/RunTests.py
@@ -9,6 +9,7 @@ import os.path as path
 from os import chdir, environ
 from shutil import rmtree
 import subprocess
+import platform
 
 #
 
@@ -44,7 +45,11 @@ if path.isdir(BUILD_DIR):
 
 environ["MP_PERFETTO_SHOULD_BE_ON"] = "FALSE"
 
-run_command (command=f"cmake -B {BUILD_DIR}",
+# Single-config generators (Unix Makefiles, Ninja) require CMAKE_BUILD_TYPE at configure time
+# Multi-config generators (Xcode, Visual Studio) ignore it and use --config at build time
+build_type_flag = "" if platform.system() == "Windows" else " -DCMAKE_BUILD_TYPE=Debug"
+
+run_command (command=f"cmake -B {BUILD_DIR}{build_type_flag}",
              workingDir=REPO_ROOT)
 
 run_command (command=f"cmake --build {BUILD_DIR}",
@@ -55,7 +60,7 @@ run_command (command="ctest -C Debug",
 
 environ["MP_PERFETTO_SHOULD_BE_ON"] = "TRUE"
 
-run_command (command=f"cmake -B {BUILD_DIR} -D PERFETTO=ON",
+run_command (command=f"cmake -B {BUILD_DIR} -D PERFETTO=ON{build_type_flag}",
              workingDir=REPO_ROOT)
 
 run_command (command=f"cmake --build {BUILD_DIR}",

--- a/scripts/RunTests.py
+++ b/scripts/RunTests.py
@@ -48,24 +48,32 @@ environ["MP_PERFETTO_SHOULD_BE_ON"] = "FALSE"
 
 # Single-config generators (Unix Makefiles, Ninja) require CMAKE_BUILD_TYPE at configure time
 # Multi-config generators (Xcode, Visual Studio) ignore it and use --config at build time
-build_type_flag = "" if platform.system() == "Windows" else " -DCMAKE_BUILD_TYPE=Debug"
+if platform.system() == "Windows":
+    build_type_flag = ""
+    generator_flag = ""
+elif platform.system() == "Linux":
+    build_type_flag = " -DCMAKE_BUILD_TYPE=Debug"
+    generator_flag = " -G Ninja"  # Ninja is faster and installed in CI
+else:
+    build_type_flag = " -DCMAKE_BUILD_TYPE=Debug"
+    generator_flag = ""
 
-run_command (command=f"cmake -B {BUILD_DIR}{build_type_flag}",
+run_command (command=f"cmake -B {BUILD_DIR}{generator_flag}{build_type_flag}",
              workingDir=REPO_ROOT)
 
-run_command (command=f"cmake --build {BUILD_DIR}",
+run_command (command=f"cmake --build {BUILD_DIR} --parallel",
              workingDir=REPO_ROOT)
 
-run_command (command="ctest -C Debug --output-on-failure",
+run_command (command="ctest -C Debug -j --output-on-failure",
              workingDir=BUILD_DIR)
 
 environ["MP_PERFETTO_SHOULD_BE_ON"] = "TRUE"
 
-run_command (command=f"cmake -B {BUILD_DIR} -D PERFETTO=ON{build_type_flag}",
+run_command (command=f"cmake -B {BUILD_DIR} -D PERFETTO=ON{generator_flag}{build_type_flag}",
              workingDir=REPO_ROOT)
 
-run_command (command=f"cmake --build {BUILD_DIR}",
+run_command (command=f"cmake --build {BUILD_DIR} --parallel",
              workingDir=REPO_ROOT)
 
-run_command (command="ctest -C Debug --output-on-failure",
+run_command (command="ctest -C Debug -j --output-on-failure",
              workingDir=BUILD_DIR)

--- a/tests/AddSubdirectory/AddTests.cmake
+++ b/tests/AddSubdirectory/AddTests.cmake
@@ -20,7 +20,8 @@ set_tests_properties ("${base_name}.configure"
 add_test (NAME "${base_name}.build"
 		  COMMAND "${CMAKE_COMMAND}"
 		  			--build "${binary_dir}"
-		  			--config "$<CONFIG>")
+		  			--config "$<CONFIG>"
+		  			--parallel)
 
 set_tests_properties ("${base_name}.build" PROPERTIES FIXTURES_REQUIRED MelatoninPerfettoAddSubdirectoryConfigure)
 

--- a/tests/AddSubdirectory/AddTests.cmake
+++ b/tests/AddSubdirectory/AddTests.cmake
@@ -6,6 +6,8 @@ add_test (NAME "${base_name}.configure"
 		  COMMAND "${CMAKE_COMMAND}"
 		  			-S "${CMAKE_CURRENT_LIST_DIR}"
 		  			-B "${binary_dir}"
+		  			-G "${CMAKE_GENERATOR}"
+		  			-D "CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
 		  			-D "MELATONIN_PERFETTO_ROOT=${MelatoninPerfetto_SOURCE_DIR}"
 		  			-D "FETCHCONTENT_SOURCE_DIR_JUCE=${juce_SOURCE_DIR}"
 		  			-D "FETCHCONTENT_SOURCE_DIR_PERFETTO=${perfetto_SOURCE_DIR}")

--- a/tests/AddSubdirectory/CMakeLists.txt
+++ b/tests/AddSubdirectory/CMakeLists.txt
@@ -26,4 +26,6 @@ juce_add_console_app (test VERSION 1.0.0)
 
 target_sources (test PRIVATE main.cpp)
 
+target_compile_definitions (test PRIVATE JUCE_USE_CURL=0)
+
 target_link_libraries (test PRIVATE Melatonin::Perfetto)

--- a/tests/DumpFiles/CMakeLists.txt
+++ b/tests/DumpFiles/CMakeLists.txt
@@ -2,6 +2,8 @@ add_executable (MelatoninDumpFileTest)
 
 target_sources (MelatoninDumpFileTest PRIVATE main.cpp)
 
+target_compile_definitions (MelatoninDumpFileTest PRIVATE JUCE_USE_CURL=0)
+
 target_link_libraries (MelatoninDumpFileTest PRIVATE Melatonin::Perfetto)
 
 add_test (NAME melatonin.perfetto.dump_file_creation

--- a/tests/FindPackage/AddTests.cmake
+++ b/tests/FindPackage/AddTests.cmake
@@ -17,6 +17,7 @@ add_test (NAME "${base_name}.configure"
 		  			-S "${CMAKE_CURRENT_LIST_DIR}"
 		  			-B "${binary_dir}"
 		  			-G "${CMAKE_GENERATOR}"
+		  			-D "CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
 		  			-D "MelatoninPerfetto_DIR=${prefix}/${CMAKE_INSTALL_LIBDIR}/cmake/melatonin_perfetto"
 		  			-D "FETCHCONTENT_SOURCE_DIR_JUCE=${juce_SOURCE_DIR}"
 		  			-D "FETCHCONTENT_SOURCE_DIR_PERFETTO=${perfetto_SOURCE_DIR}")

--- a/tests/FindPackage/AddTests.cmake
+++ b/tests/FindPackage/AddTests.cmake
@@ -28,9 +28,10 @@ set_tests_properties ("${base_name}.configure" PROPERTIES
 					  ENVIRONMENT_MODIFICATION "MP_PERFETTO_SHOULD_BE_ON=unset:")
 
 add_test (NAME "${base_name}.build"
-		  COMMAND "${CMAKE_COMMAND}" 
+		  COMMAND "${CMAKE_COMMAND}"
 		  			--build "${binary_dir}"
-		  			--config "$<CONFIG>")
+		  			--config "$<CONFIG>"
+		  			--parallel)
 
 set_tests_properties ("${base_name}.build" PROPERTIES FIXTURES_REQUIRED MelatoninPerfettoFindPackageConfigure)
 

--- a/tests/FindPackage/CMakeLists.txt
+++ b/tests/FindPackage/CMakeLists.txt
@@ -22,4 +22,6 @@ juce_add_console_app (test VERSION 1.0.0)
 
 target_sources (test PRIVATE main.cpp)
 
+target_compile_definitions (test PRIVATE JUCE_USE_CURL=0)
+
 target_link_libraries (test PRIVATE Melatonin::Perfetto)

--- a/tests/PerfettoValue/CMakeLists.txt
+++ b/tests/PerfettoValue/CMakeLists.txt
@@ -6,6 +6,8 @@ else()
 	target_sources (PerfettoValueTest PRIVATE should_be_off.cpp)
 endif()
 
+target_compile_definitions (PerfettoValueTest PRIVATE JUCE_USE_CURL=0)
+
 target_link_libraries (PerfettoValueTest PRIVATE Melatonin::Perfetto)
 
 add_test (NAME melatonin.perfetto.PERFETTO_symbol_value

